packages/jest-environment/src/index.ts

```diff
@@ -139,41 +164,42 @@ export interface Jest {
   /**
    * Mocks a module with the provided module factory when it is being imported.
    */
-  unstable_mockModule<T = unknown>(
+  unstable_mockModule(
     moduleName: string,
-    moduleFactory: () => Promise<T> | T,
+    moduleFactory: () => Promise<unknown> | unknown,
```

Comment:
Is the generic necessary here? Seems like not. Or I missed something?

Reply:
it allows people to do `jest.unstable_mockModule<typeof import('fs')>(() => {})` and have the return value be typed

Reply:
Alright, I will revert. Should something similar be added to `jest.mock()`, `jest.requireActual()` and similar? `@types/jest` has these generics. I wasn’t sure.

Reply:
Yeah, I think so